### PR TITLE
Added a dns test

### PR DESCRIPTION
**Pull request by jesings**
This is going to fail CI until the other 2 fix branches in native client and rustposix are merged, we should rerun it at that point

### CODE REVIEW

**Review by kapkic — APPROVED** [review]
Looks good!

File: tests/test_cases/dettests.txt [hunk 1]
```diff
@@ -26,3 +26,4 @@ getuid.c
 forkexecuid.c
 ioctl.c
 setsid.c
+dnstest.c
```
Comment (kapkic) [inline tests/test_cases/dettests.txt:29]: We can also add popen.c here too, since we already have a test for it.

Comment (rennergade) [reply tests/test_cases/dettests.txt:29]: It's more complicated because you need the shell loaded as well.

**Review by rennergade — APPROVED** [review]
Looks good!

**Comment by rennergade** [comment]
@jesings this actually seems to be failing the test suite?

**Comment by rennergade** [comment]
Oh maybe because the other branches havent updated yet.

**Pull request merged**